lib/ui/painting/immutable_buffer.cc

```diff
@@ -53,6 +58,49 @@ size_t ImmutableBuffer::GetAllocationSize() const {
   return sizeof(ImmutableBuffer) + data_->size();
 }
 
+void ImmutableBuffer::fromAsset(Dart_NativeArguments args) {
+  UIDartState::ThrowIfUIOperationsProhibited();
+  Dart_Handle callback_handle = Dart_GetNativeArgument(args, 2);
+  if (!Dart_IsClosure(callback_handle)) {
+    Dart_SetReturnValue(args, tonic::ToDart("Callback must be a function"));
+    return;
+  }
+  Dart_Handle asset_name_handle = Dart_GetNativeArgument(args, 0);
+  uint8_t* chars = nullptr;
+  intptr_t asset_length = 0;
+  Dart_Handle result =
+      Dart_StringToUTF8(asset_name_handle, &chars, &asset_length);
+  if (Dart_IsError(result)) {
+    Dart_PropagateError(result);
+    return;
+  }
+  Dart_Handle immutable_buffer = Dart_GetNativeArgument(args, 1);
+
+  std::string asset_name = std::string{reinterpret_cast<const char*>(chars),
+                                       static_cast<size_t>(asset_length)};
+
+  std::shared_ptr<AssetManager> asset_manager = UIDartState::Current()
+                                                    ->platform_configuration()
+                                                    ->client()
+                                                    ->GetAssetManager();
+  std::unique_ptr<fml::Mapping> data = asset_manager->GetAsMapping(asset_name);
+
+  if (data == nullptr) {
+    return;
```

Comment:
Return a Dart String here with something like "Asset not found".

Reply:
Well, Dart_SetReturnValue

Reply:
The contract for asset loading is already that we return null if it isn't found, and the framework can turn this into an exception